test/libsolidity/smtCheckerTests/inheritance/constructor_state_variable_init_chain_tree.sol

```diff
@@ -0,0 +1,43 @@
+pragma experimental SMTChecker;
+
+contract A {
+	int x;
+	constructor (int a) { x = a; }
+}
+
+contract Z {
+	int z;
+	constructor(int _z) {
+		z = _z;
+	}
+}
+
+contract B is A, Z {
+	constructor(int b) A(b) Z(x) {
+		assert(x == b);
+		assert(z == b);
+	}
+}
+
+contract F is Z, A {
+	constructor(int b) Z(x) A(b) {
+		assert(x == b);
+		assert(z == 0);
+	}
+}
+
+contract C is B {
+	constructor(int c) B(-c) {
+		if (x > 0) {
+			assert(c < 0); // should hold
```

Comment:
It holds here and fails on next line?

Reply:
Yes, should be. Anything wrong?

Reply:
No, I was thinking that because of comment you wrote "should fail" that it maybe doesn't fail. All fine :)